tensorrt_llm/_torch/pyexecutor/llm_request.py

```diff
@@ -108,6 +140,44 @@ def get(self, all_logits: bool) -> torch.Tensor | None:
         except IndexError:
             return None
 
+    def _add_fragment(self, logits: torch.Tensor):
+        """Add a logits fragment to device storage"""
+        self._device_fragments.append(logits.clone())
+
+        # Streaming mode: transfer immediately after each fragment (self.chunk_size=1).
+        # Non-streaming mode: batch transfer every chunk_size steps.
+        if len(self._device_fragments) == self.chunk_size:
+            self._transfer_chunk_to_host()
+
+    def _transfer_chunk_to_host(self):
+        """Transfer accumulated fragments to host"""
+        if not self._device_fragments:
+            return
+
+        # Allocate host storage if needed
+        assert self._storage is not None, "Storage should be initialized"
+        # if self._storage is None:
+        #     self._init(self._device_fragments[0])
+
+        # Merge fragments on device first
+        merged_logits = torch.cat(self._device_fragments, dim=0)
+
+        # Copy to host (device-to-host transfer)
+        end_pos = self._current_position + len(self._device_fragments)
+        self._storage[self._current_position:end_pos].copy_(merged_logits,
+                                                            non_blocking=True)
+
+        # Update position and clear fragments
+        self._logits_indices.append((self._current_position, end_pos))
+        self._current_position = end_pos
+        self._device_fragments.clear()
+
+    def finalize_transfer(self):
+        """Force transfer of any remaining fragments to host (for chunked mode)"""
+        if self.use_chunked_logits and hasattr(
+                self, '_device_fragments') and self._device_fragments:
+            self._transfer_chunk_to_host()
```

Comment:
technically `finalize_transfer` can be absorbed inside `self._transfer_chunk_to_host()` and that can be called directly in `PyResult`  right? might simplify things a bit?